pkg/cri/config/config.go

```diff
@@ -156,7 +156,7 @@ type CniConfig struct {
 	// (https://kubernetes.io/docs/concepts/cluster-administration/network-plugins/#kubenet)
 	// today, who don't have a cni daemonset in production. NetworkPluginConfTemplate is
 	// a temporary backward-compatible solution for them.
-	// TODO(random-liu): Deprecate this option when kubenet is deprecated.
+	// DEPRECATED: use CNI configs
```

Comment:
Substantially we have been deprecating this for years, but looks like we have forgotten to formally deprecate it

Reply:
Commented on https://github.com/containerd/containerd/issues/8596. It's unfortunate this was ostensibly deprecated for so long without any notice, I don't think a reasonable replacement is readily available yet. This allowed for very simple generic networking using the Kubernetes built in node pod CIDR IPAM.

Reply:
This TODO is from a time where this field was going to be deprecated and made sense at that context https://github.com/kubernetes/kubernetes/issues/57130

However, the time evolved and showed to be useful and there are several projects dependent on it , as a proof you can see that the kubernetes field got added new features instead of being deprecated as you can see in this KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2593-multiple-cluster-cidrs

So, it seems that happened here is that we forget to tell containerd how useful is this field and this TODO should have been removed

Reply:
Sure, let's de-deprecate this